src/encoded/static/components/navigation/components/SearchBar.js

```diff
@@ -147,45 +207,58 @@ export class SearchBar extends React.PureComponent{
-            <form className={_.filter(formClasses).join(' ')} action={searchAllItems ? "/search/" : "/browse/" } method="GET">
-                <SelectItemTypeDropdownBtn currentAction={currentAction} visible={!!(searchBoxHasInput || searchQueryFromHref)}
-                    toggleSearchAllItems={this.toggleSearchAllItems} searchAllItems={searchAllItems} />
+            <form className={_.filter(formClasses).join(' ')} action={this.getActionType()} method="GET">
+                <SelectItemTypeDropdownBtn currentAction={currentAction} href={href} visible={!!(searchBoxHasInput || searchQueryFromHref)}
+                    toggleSearchAllItems={this.toggleSearchAllItems} getDropdownTitleText={this.getDropdownTitleText} currentPageType={this.currentPageType} searchItemType={searchItemType} />
                 <input className="form-control search-query" id="navbar-search" type="search" placeholder="Search"
                     name="q" value={typedSearchQuery} onChange={this.onSearchInputChange} key="search-input" onBlur={this.onSearchInputBlur} />
-                { showingCurrentQuery ? <i className="reset-button icon icon-times fas" onClick={this.onResetSearch}/> : null }
-                { showingCurrentQuery ? null : (
+                {showingCurrentQuery ? <i className="reset-button icon icon-times fas" onClick={this.onResetSearch} /> : null}
+                {showingCurrentQuery ? null : (
                     <button type="submit" className="search-icon-button">
-                        <i className="icon icon-fw icon-search fas"/>
+                        <i className="icon icon-fw icon-search fas" />
                     </button>
-                ) }
-                { SearchBar.renderHiddenInputsForURIQuery(query) }
+                )}
+                {SearchBar.renderHiddenInputsForURIQuery(query)}
             </form>
         );
     }
 }
 
-const SelectItemTypeDropdownBtn = React.memo(function SelectItemTypeDropdownBtn(props){
-    const { currentAction, searchAllItems, toggleSearchAllItems, visible } = props;
+const SelectItemTypeDropdownBtn = React.memo(function SelectItemTypeDropdownBtn(props) {
+    const { currentAction, toggleSearchAllItems, visible, getDropdownTitleText, href, searchItemType } = props;
+
     if (isSelectAction(currentAction) || !visible) return null;
+
     return (
         <Fade in={visible} appear>
             <div className="search-item-type-wrapper">
                 <DropdownButton id="search-item-type-selector" size="sm" variant="outline-secondary"
-                    onSelect={(eventKey, evt)=>{ toggleSearchAllItems(eventKey === 'all' ? true : false); }}
-                    title={searchAllItems ? 'All Items' : 'Experiment Sets'}>
-                    <DropdownItem eventKey="sets" data-key="sets" active={!searchAllItems}>
+                    onSelect={(eventKey, evt) => { toggleSearchAllItems(eventKey); }}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JSX props should not use arrow functions (react/jsx-no-bind)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4513406)